airgun/views/host_new.py

```diff
@@ -418,3 +419,71 @@ class RecurringJobDialog(Pf4ConfirmationDialog):
 
     confirm_dialog = Button(locator='.//button[@data-ouia-component-id="btn-modal-confirm"]')
     cancel_dialog = Button(locator='.//button[@data-ouia-component-id="btn-modal-cancel"]')
+
+
+class PF4CheckboxTreeView(CheckboxGroup):
+    """
+    Modified :class:`airgun.widgets.CheckboxGroup` for PF4 tree view with checkboxes:
+        https://www.patternfly.org/v4/components/tree-view#with-checkboxes
+    """
+    ITEMS_LOCATOR = './/label'
+    CHECKBOX_LOCATOR = './/label[normalize-space(.)="{}"]/preceding-sibling::span/input[@type="checkbox"]'
+
+
+class ManageColumnsView(BaseLoggedInView):
+    """ Manage columns modal. """
+    ROOT = '//div[contains(@class, "pf-c-modal-box")]'
+
+    CHECKBOX_SECTION_TOGGLE = './/label[normalize-space(.)="{}"]/preceding-sibling::button'
+    DEFAULT_COLLAPSED_SECTIONS = [
+        CHECKBOX_SECTION_TOGGLE.format('Content'),
+        CHECKBOX_SECTION_TOGGLE.format('Network'),
+        CHECKBOX_SECTION_TOGGLE.format('Reported data'),
+        CHECKBOX_SECTION_TOGGLE.format('RH Cloud')
+    ]
+    is_tree_collapsed = True
+    title = Text('.//header//span[contains(@class, "pf-c-modal-box__title")][normalize-space(.)="Manage columns"]')
+    confirm_dialog = Button(locator='.//button[normalize-space(.)="Save"]')
+    cancel_dialog = Button(locator='.//button[normalize-space(.)="Cancel"]')
+    checkbox_group = PF4CheckboxTreeView(locator='.//div[contains(@class, "pf-c-tree-view")]')
+
+    def collapsed_sections(self):
```

Comment:
Does the return value ever change? Can't this just be a class variable?

Reply:
The value doesn't change, but I cannot define Widget instances in a list in the class (at least if failed for me). So I defined only the locators and then I return the corresponding web elements here.
However, as I mentioned in the above comment, this is kinda a helper for the PF4CheckboxTreeView functionality and it will be refactored in a stand-alone widget under `airgun/widgets.py`.